### DISCUSSION TITLE
A few general Question

### DISCUSSION THREAD

**Discussion by sLesage** [HashLoad/boss#148] [Q&A] [answered]
Hi,

A colleague of mine showed me the BOSS Package Manager, and I have been trying to get it working in our environment here. Now ... for my understanding ... I seem to be having a few questions to which I hope someone here can give me a good answer.

- Version numbers can only have 3 sets of digits right ? I tried with 1.0.0.1 but that gave me errors. Fine with me if only 3 digits are used though, I just need to take that into account.
- Do version numbers work on any branch ? I mean we have develop and master / main. So if i add 1.0.3 as a tag to main and 1.0.4 to develop using boss install <repo>:1.0.3 will get the version tagged with 1.0.3. (main) right ? And boss install <repo>:1.0.4 will get that version from develop.
- How do you make sure it uses the last commit on main (and not the last one on develop) ?
- Is there something like a boss install <repo>:latest to get the last verstion / commit from Master ?
- If my project currently needs 1.0.3 but I need to make it use 1.0.4 I can simply use boss install <repo>:1.0.4 right ?
- boss -install simply sets up the boss dependencies right ? ... in order to get the correct versions of those dependencies I always have to use boss -update ?
- is there any way to change the default path for sources ? It's now using ./src as default but is it possible to change that to ./sources instead ?
- Is there any way to trigger a boss -update after every fetch / pull from the Git reporistory ?

I'm really looking forward to using BOSS in our setup, because it would solve some dependency issues we currently have. But as you can see, there is probably still a lot I don't fully understand, so it would be awesome if someone would have the ansers, or if we could add this to the documentation as well.

**Comment by sbeu** [comment, marked as answer]
1) Version numbers: Yes, we need to use only 3 digits
2) I think a TAG is not linked to a branch. It is linked to a commit.
3) When you create a tag, you select the branch on which you create a tag (at least in Azure Devops) and then the tag created uses the latest current commit on this branch. I think you can also select any commit to create a tag.
4) I don't think so. I only use tags, and I think it is the only way to identify a version for boss
5) I don't know if we can use boss like that "boss install:1.0.4"
I use boss.json file. In this file you can specify the version you want to fetch (with a simple "boss update")
6) "boss install" initializes the boss.json file
"boss update" reads the boss.json file and fetches the needed dependencies
7) Yes it is possible. In the boss.json file of the package, you can specify the location of the sources to be distributed
8) Maybe with native git. I use Fork and I can add custom actions after a pull

Actually I figured out all these answers reading at the documentation :)
